include/aws/iotdevice/secure_tunneling.h

```diff
@@ -53,6 +54,7 @@ struct aws_secure_tunneling_connection_config {
     struct aws_allocator *allocator;
     struct aws_client_bootstrap *bootstrap;
     struct aws_socket_options *socket_options;
+    struct aws_http_proxy_options *http_proxy_options;
```

Comment:
This isn't a structure that can be stored persistently as it contains a number of references to external things (byte cursors).  The persistent version is aws_http_proxy_config which can be created from the options struct.